Download only one ffmpeg

jaruba opened the pull request:
- removed 32bit binaries for Linux (who even uses 32bit Linux anymore?)
- split `ffmpeg.tgz` into: `ffmpeg-linux.tgz`, `ffmpeg-darwin.tgz`, `ffmpeg-win32.tgz`
- now downloads only the `ffmpeg` binary for your platform

It's currently hooked up to a repo where I released the 3 separate archives, if you choose to accept this PR you should change the info accordingly to download from your repo. (check comments from `install.js`)

As a side note, I saw your using `.setFfmpegPath()`, is `setFfprobePath()` not required? Does it check for it in the same folder as `ffmpeg` anyway?

jaruba commented:
I've had a chat with @acidhax about this PR. He mentioned that the best approach might be to use only the 32bit on Linux.

Says 64bit `ffmpeg` has no performance benefits over 32bit, while 32bit will be accepted by all architectures. Thoughts?

konsumer commented:
I am cool with 32, if it works reliably on 64 linux (testing needed.) I think we should also do a fallback, if ffmpeg is in your path, and not even download it.

`.setFfmpegPath()` alone works fine for regular use, but I'm not managing probe, so even if we set the path, it wouldn't work on some platforms (if my memory serves me, not all platforms have pre-built binaries available.) Again, local path fallback would help with this...

jaruba commented:
> `.setFfmpegPath()` alone works fine for regular use, but I'm not managing probe, so even if we set the path, it wouldn't work on some platforms (if my memory serves me, not all platforms have pre-built binaries available.)

You have `ffprobe` for all platforms in your archives too. I've used `ffprobe` in one of my projects on OSX, Linux and Windows months ago, so I'm sure it works and that there are prebuilts available for all platforms.

**Changes to the PR:**
- download `ffmpeg` package only if it is not already installed
- changed to 32bit `ffmpeg` on Linux (I've also changed the archive from my test repo from 64bit to 32bit so it will download the correct architecture for tests)
- added support for `ffprobe` too (by setting with `.setFfprobePath()` and checking if it is already downloaded in `install.js`)

**Notes:**
- both `ffmpeg` and `ffprobe` need to be installed in order to not download the prebuilts
- what if the installed `ffmpeg` and `ffprobe` from the user's computer are not the same versions as the ones that the author of the module / app that is using `easy-ffmpeg` as a dependency tested on?

I'll still need to test it, but the code looks alright, please review the code changes when you have time.

jaruba commented:
I've tested it on Windows now.

```
npm install jaruba/easy-ffmpeg
```

**With ffmpeg/ffprobe already installed:**

![folder1](https://cloud.githubusercontent.com/assets/1777923/13666610/12254f90-e6bb-11e5-8586-b2c46c4520c2.png)

![check-paths1](https://cloud.githubusercontent.com/assets/1777923/13666615/188879b6-e6bb-11e5-8804-3f327315fd59.png)

**Without ffmpeg/ffprobe installed:**

![folder2](https://cloud.githubusercontent.com/assets/1777923/13666620/2682a802-e6bb-11e5-81b9-148332f4ad69.png)

![check-paths2](https://cloud.githubusercontent.com/assets/1777923/13666625/2cbadc3a-e6bb-11e5-90bf-76688f305ce4.png)

It looks like it's working fine to me..

konsumer commented:
Nice. when I have some time, I will check on all supported platforms. As a sidenote, what is that tool you are using to get a chrome devtools-like console for nodejs code? Is that atomshell/nodewebkit?

jaruba commented:
> I will check on all supported platforms

Thanks!

> what is that tool you are using to get a chrome devtools-like console for nodejs code? Is that atomshell/nodewebkit?

Yes, it's the devtools from `NW.js` (node-webkit).

jaruba commented:
Tested on OSX, works perfectly.

konsumer commented:
Looks good. Thanks for doing all the testing!

konsumer commented:
I'll bump npm and test on Linux when I'm not on my phone.

jaruba commented:
great!

konsumer commented:
hmm. I get this error when the postinstall runs (node install.js):

```
> easy-ffmpeg@0.0.10 postinstall /Users/konsumer/Desktop/easy-ffmpeg
> node install.js

events.js:154
      throw er; // Unhandled 'error' event
      ^

Error: incorrect header check
    at Zlib._handle.onerror (zlib.js:363:17)
```

El Capitan
node v5.5.0
npm 3.5.3

konsumer commented:
Ah, sorry. I need to run getbin and put them up on releases.

konsumer commented:
Did you actually test getbin.sh to build the files for releases? It seems like getbin & install.js have totally wrong paths in them. I will fix them up, now.

jaruba commented:
Did not test `getbin.sh`, I made the packages on windows with 7zip, said it works on OSX because I included my fork of `easy-ffmpeg` in a demo app, gave it to a friend on OSX and it worked.

But come to think about it, the way this module works now, if he had ffmpeg and ffprobe installed already it might not have downloaded the package at all for him. :/

konsumer commented:
hehe. no prob. I'm fixing it up, now.  I appreciate your help.

konsumer commented:
ok, updated and published to npm.
